packages/wonder-blocks-dropdown/src/components/__tests__/multi-select.test.tsx

```diff
@@ -3420,4 +3420,115 @@ describe("MultiSelect", () => {
             });
         });
     });
+
+    describe("readOnly prop", () => {
+        describe.each([
+            {name: "default", opener: undefined},
+            {name: "custom", opener: () => <div>Custom opener</div>},
+        ])("With $name opener", ({opener}) => {
+            it("should set aria-disabled to true if readOnly is true", () => {
+                // Arrange
+                // Act
+                doRender(
+                    <MultiSelect
+                        readOnly={true}
+                        onChange={jest.fn()}
+                        opener={opener}
+                    >
+                        <OptionItem label="item 1" value="1" />
+                        <OptionItem label="item 2" value="2" />
+                    </MultiSelect>,
+                );
+
+                // Assert
+                expect(screen.getByRole("combobox")).toHaveAttribute(
+                    "aria-disabled",
+                    "true",
+                );
+            });
+
+            it("should not have an open dropdown if readOnly is true and opened is true", () => {
+                // Arrange
+                // Act
+                doRender(
+                    <MultiSelect
+                        readOnly={true}
+                        opened={true}
+                        onChange={jest.fn()}
+                        opener={opener}
+                    >
+                        <OptionItem label="item 1" value="1" />
+                        <OptionItem label="item 2" value="2" />
+                    </MultiSelect>,
+                );
+
+                // Assert
+                expect(screen.queryByRole("listbox")).not.toBeInTheDocument();
+            });
+
+            it("should not open the dropdown if readOnly is true and the opener is clicked", async () => {
+                // Arrange
+                const {userEvent} = doRender(
+                    <MultiSelect
+                        readOnly={true}
+                        onChange={jest.fn()}
+                        opener={opener}
+                    >
+                        <OptionItem label="item 1" value="1" />
+                        <OptionItem label="item 2" value="2" />
+                    </MultiSelect>,
+                );
+                const combobox = await screen.findByRole("combobox");
```

Comment:
I remembered today that Jeff suggested we avoid `findBy` queries unless absolutely necessary since they return Promises. Does this test pass with synchronous `screen.getByRole`?